Backpack/src/main/java/net/skyscanner/backpack/calendar/presenter/BpkCalendarController.kt

```diff
@@ -0,0 +1,56 @@
+package net.skyscanner.backpack.calendar.presenter
+
+import net.skyscanner.backpack.calendar.model.CalendarDay
+import net.skyscanner.backpack.calendar.model.CalendarRange
+import java.text.SimpleDateFormat
+import java.util.*
+
+abstract class BpkCalendarController {
+  abstract val selectedDay: CalendarDay
+
+  abstract val range: CalendarRange
+
+  abstract val minDate: Calendar
+
+  abstract val maxDate: Calendar
+
+  abstract val isRtl: Boolean
+
+  abstract val locale: Locale
+
+  abstract fun onRangeSelected(range: CalendarRange)
```

Comment:
Isn't there too many abstract properties and methods here? This forces the client to override all of them and write too many boilerplate code (while they only actually use a subset of them). I think a better API design would be to provide sensible defaults, like `Date()` (today) for `minDate`, empty function for `onRangeSelected` and so on.

Reply:
Good idea! I refactored it a bit, so hopefully it's more clear now!